File: doc-testing/src/sdk_openai_agents.ts
import "dotenv/config";
import { init, openaiAgentsProcessor, shutdown } from "neatlogs";
import { addTraceProcessor, Agent, run } from "@openai/agents";

async function main() {
  await init({
    apiKey: process.env.NEATLOGS_API_KEY,
    workflowName: "openai-agents-demo",
  });

  addTraceProcessor(openaiAgentsProcessor());

  const agent = new Agent({
    name: "assistant",
    instructions: "Be concise.",
  });

  const result = await run(
    agent,
    "In one sentence, what is the OpenAI Agents SDK?",
  );
  console.log(result.finalOutput);

  await shutdown();
}

main().catch(console.error);

File: agents/src/agents/sdk_openai_agents.py
import os

from dotenv import load_dotenv
from agents import Agent, Runner, add_trace_processor
import neatlogs

load_dotenv()

neatlogs.init(api_key=os.environ["NEATLOGS_API_KEY"], endpoint=os.environ.get("NEATLOGS_ENDPOINT", "http://localhost:4100"), workflow_name="openai-agents-demo")
add_trace_processor(neatlogs.openai_agents_processor())

agent = Agent(name="assistant", instructions="Be concise.")
result = Runner.run_sync(agent, "In one sentence, what is the OpenAI Agents SDK?")
print(result.final_output)

neatlogs.flush()
neatlogs.shutdown()
